### PR TITLE
Add comprehensive UI test suite for screens and navigation

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
## Summary
This PR adds a comprehensive test suite covering the main UI screens and navigation components of the app. The tests use Robolectric and Compose testing utilities to verify screen rendering, user interactions, and callback invocations.

## Key Changes

- **AddWordScreenContentTest** (549 lines): Tests for the Add Word screen covering:
  - Basic rendering and navigation
  - AI toggle and translation field visibility logic
  - Translation direction row behavior
  - Preview and Add button states and interactions
  - Pending words list rendering and deletion
  - Error and success message display
  - Preview and existing word dialog wiring

- **WordListScreenTest** (263 lines): Tests for the Word List screen covering:
  - Empty state and word filtering (case-insensitive substring matching)
  - Search query handling and whitespace trimming
  - Item menu actions (edit, reset, delete)
  - Edit dialog validation (prevents saving blank words)
  - Reset and delete confirmation dialogs

- **AddWordDialogsTest** (198 lines): Tests for dialog components:
  - AddWordPreviewDialog display and interaction
  - ExistingWordDialog display and interaction
  - Loading states and button disabling during async operations

- **DojoScreenTest** (186 lines): Tests for the Dojo (learning) screen covering:
  - Loading and empty states
  - Word and translation display
  - Answer reveal functionality
  - Difficulty rating button interactions
  - Stats header display
  - Undo functionality and confirmation snackbar

- **BottomNavigationBarTest** (150 lines): Tests for navigation:
  - Tab selection and navigation to destinations
  - Route correctness for all screens
  - Re-selection behavior and back stack management

- **AppsListScreenContentTest** (136 lines): Tests for the Apps List screen covering:
  - Loading and error states
  - App list rendering with selection state
  - App toggle and enable/disable functionality

- **ScreenTest** (38 lines): Unit tests for navigation routes:
  - Route uniqueness and non-emptiness
  - Expected route values for all screens

- **AppsListScreen.kt**: Made `AppsListScreenContent` composable `internal` to enable testing

## Implementation Details

- All tests use `@RunWith(RobolectricTestRunner::class)` with SDK 33 and xlarge qualifiers
- Tests leverage `ComponentActivityRegistrationRule` for proper Compose test setup
- Mock callbacks use `mockk(relaxed = true)` for verification
- Tests verify both UI state rendering and callback invocations
- Search and filtering logic is tested with edge cases (whitespace, case-insensitivity)
- Dialog interactions are tested with loading states and disabled button scenarios

https://claude.ai/code/session_01Sm4q6vpUt1XpoLyd3sj3n7

### CODE REVIEW

File: app/src/test/java/com/procrastilearn/app/navigation/BottomNavigationBarTest.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(
+    sdk = [33],
+    manifest = Config.NONE,
+    qualifiers = "xlarge",
+)
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/navigation/BottomNavigationBarTest.kt]: remove this

File: app/src/test/java/com/procrastilearn/app/ui/dojo/DojoScreenTest.kt [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(
+    sdk = [33],
+    manifest = Config.NONE,
+    qualifiers = "xlarge",
+)
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/dojo/DojoScreenTest.kt]: remove this

File: app/src/test/java/com/procrastilearn/app/ui/screens/AddWordDialogsTest.kt [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(
+    sdk = [33],
+    manifest = Config.NONE,
+    qualifiers = "xlarge",
+)
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/AddWordDialogsTest.kt]: remove this

File: app/src/test/java/com/procrastilearn/app/ui/screens/AddWordDialogsTest.kt [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    @Before
+    fun setup() {
+        onCancel = mockk(relaxed = true)
+        onConfirm = mockk(relaxed = true)
+        context = ApplicationProvider.getApplicationContext()
+    }
+
+    private fun string(resId: Int) = context.getString(resId)
+
+    // --- AddWordPreviewDialog ---
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/AddWordDialogsTest.kt]: remove this and similar comments

File: app/src/test/java/com/procrastilearn/app/ui/screens/AddWordScreenContentTest.kt [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(
+    sdk = [33],
+    manifest = Config.NONE,
+    qualifiers = "xlarge",
+)
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/AddWordScreenContentTest.kt]: remove this

File: app/src/test/java/com/procrastilearn/app/ui/screens/AddWordScreenContentTest.kt [hunk 6]
```diff
@@ -0,0 +1,549 @@
...
+        onPreviewCancel = mockk(relaxed = true)
+        onPreviewConfirmAdd = mockk(relaxed = true)
+        onAddClick = mockk(relaxed = true)
+        onExistingWordDialogCancel = mockk(relaxed = true)
+        onExistingWordDialogProceed = mockk(relaxed = true)
+    }
+
+    private fun string(resId: Int) = context.getString(resId)
+
+    // --- Basic rendering & navigation ---
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/AddWordScreenContentTest.kt]: same, remove comments

File: app/src/test/java/com/procrastilearn/app/ui/screens/AppsListScreenContentTest.kt [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(
+    sdk = [33],
+    manifest = Config.NONE,
+    qualifiers = "xlarge",
+)
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/AppsListScreenContentTest.kt]: same

File: app/src/test/java/com/procrastilearn/app/ui/screens/WordListScreenTest.kt [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(
+    sdk = [33],
+    manifest = Config.NONE,
+    qualifiers = "xlarge",
+)
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/WordListScreenTest.kt]: same

File: app/src/test/java/com/procrastilearn/app/ui/screens/WordListScreenTest.kt [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+        context = ApplicationProvider.getApplicationContext()
+        onDelete = mockk(relaxed = true)
+        onEdit = mockk(relaxed = true)
+        onReset = mockk(relaxed = true)
+        onNavigateBack = mockk(relaxed = true)
+    }
+
+    private fun string(resId: Int) = context.getString(resId)
+
+    // --- WordListContent ---
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/WordListScreenTest.kt]: same here and in other places
